packages/config/src/options.js

```diff
@@ -35,6 +35,12 @@ export function getNuxtConfig (_options) {
     options._routerBaseSpecified = true
   }
 
+  // TODO: Remove for Nuxt 3
+  // router.scrollBehavior -> app/router.scrollBehavior.js
+  if (options.router && typeof options.router !== 'undefined') {
```

Comment:
Seems the warning would display whenever `router` is defined, I think it should check if `scrollBehavior` is defined too.

Reply:
Fixed, good catch @kevinmarrec